### DISCUSSION TITLE
Tab for completion

### DISCUSSION THREAD

**Discussion by bryant-the-coder** [neoclide/coc.nvim#3457] [Q&A] [answered]
Hi, I am trying to map `<tab>` to trigger completion in my keymaps.lua Is it possible?  Does `coc-snippets` support `<tab>` completion? And like the coc-description etc? Can help? I am still new 😅

**Comment by fannheyward** [comment, marked as answer]
```lua
vim.api.nvim_set_keymap("i", "<TAB>", "pumvisible() ? '<C-n>' : '<TAB>'", {noremap = true, silent = true, expr = true, replace_keycodes = false})
```

**Reply by bryant-the-coder** [reply]
```
nnoremap <silent> K :call <SID>show_documentation()<CR>
nmap <silent> [g <Plug>(coc-diagnostic-prev)
```

What about this?

**Comment by ZaneBartlett1** [comment]
Hey since this is already put together (huge thank you!) it might be worth putting [here](https://github.com/neoclide/coc.nvim/wiki/Completion-with-sources#use-tab-or-custom-key-for-trigger-completion)? 
This isn't the first search result you get even if you put something like nvim lua tab complete coc, so I just figured it might be helpful for others. Again, thank you!
